CMakeLists.txt

```diff
@@ -236,7 +236,11 @@ elseif(CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
         endif()
     endif()
     if(MINGW)
-        list(APPEND WARNFLAGS_DISABLE -Wno-pedantic-ms-format)
+        # Add `-Wno-pedantic-ms-format` only if the toolchain is GCC MinGW-w64
```

Comment:
What are the warnings you were seeing? Can you list the compiler output? AFAIK, from a search of our GitHub code, we don't use `%I64d`.

Reply:
> What are the warnings you were seeing? Can you list the compiler output? AFAIK, from a search of our GitHub code, we don't use `%I64d` or `%I32d`

clang does not understand `-Wno-pedantic-ms-format` regardless of whether `%I64d` or `%I32d` is used.
This does not cause any harm in compiling .o files, but the warning of unknown parameters is treated as an error in linking .exe files.

This is a build log from compiling unmodified zlib-ng-2.1.3 with llvm-mingw.
```
[ 76%] Building C object test/CMakeFiles/maketrees.dir/__/zutil.c.obj
warning: unknown warning option '-Wno-pedantic-ms-format'; did you mean '-Wno-pedantic-macros'? [-Wunknown-warning-option]
1 warning generated.
[ 76%] Linking C executable ..\maketrees.exe
[ 76%] Built target maketrees
[ 76%] Building C object test/CMakeFiles/makecrct.dir/__/tools/makecrct.c.obj
warning: unknown warning option '-Wno-pedantic-ms-format'; did you mean '-Wno-pedantic-macros'? [-Wunknown-warning-option]
1 warning generated.
[ 77%] Linking C executable ..\makecrct.exe
[ 77%] Built target makecrct
[ 78%] Building CXX object _deps/googletest-build/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.obj
error: unknown warning option '-Wno-pedantic-ms-format'; did you mean '-Wno-pedantic-macros'? [-Werror,-Wunknown-warning-option]
mingw32-make[2]: *** [_deps\googletest-build\googletest\CMakeFiles\gtest.dir\build.make:76: _deps/googletest-build/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:509: _deps/googletest-build/googletest/CMakeFiles/gtest.dir/all] Error 2
mingw32-make: *** [Makefile:145: all] Error 2
```

Reply:
Here is the original PR for the `-Wno-pedantic-ms-format` change https://github.com/zlib-ng/zlib-ng/pull/1362.

Reply:
Another option would be to check if the compiler supported the flag.
```cmake
check_c_compiler_flag(-Wno-pedantic-ms-format HAVE_NO_PEDANTIC_MS_FORMAT)
if(HAVE_NO_PEDANTIC_MS_FORMAT)
    list(APPEND WARNFLAGS_DISABLE -Wno-pedantic-ms-format)
endif()
```

Reply:
I didn't know CMake had such capabilities. I will try checking it.

Reply:
Was this change also going in or not?

Reply:
Sorry, I forgot to test this last time! Fortunately, it works like a charm on both llvm-mingw and MinGW-w64 GCC.

https://github.com/zlib-ng/zlib-ng/pull/1570/commits/3d14566fa9bc042f88c879d5c2813b738990d632 introduced `check_c_compiler_flag` based `-Wno-pedantic-ms-format` detection.